filebeat/harvester/log.go

```diff
@@ -189,42 +189,36 @@ func (h *Harvester) openStdin() (encoding.Encoding, error) {
 	return h.encoding(h.file)
 }
 
+// openFile opens a file and checks for the encoding. In case the encoding cannot be detected
+// or the file cannot be opened because for example of failing read permissions, an error
+// is returned and the harvester is closed. The file will be picked up again the next time
+// the file system is scanned
 func (h *Harvester) openFile() (encoding.Encoding, error) {
 	var file *os.File
 	var err error
 	var encoding encoding.Encoding
 
-	// TODO: This is currently end endless retry, should be set to a max?
-	// retry on failure.
-	for {
-		file, err = input.ReadOpen(h.Path)
-		if err == nil {
-			// Check we are not following a rabbit hole (symlinks, etc.)
-			if !input.IsRegularFile(file) {
-				file.Close()
-				return nil, errors.New("Given file is not a regular file.")
-			}
-
-			encoding, err = h.encoding(file)
-			if err == nil {
-				break
-			}
+	file, err = input.ReadOpen(h.Path)
+	if err == nil {
+		// Check we are not following a rabbit hole (symlinks, etc.)
+		if !input.IsRegularFile(file) {
+			return nil, errors.New("Given file is not a regular file.")
+		}
 
-			file.Close()
-			if err != transform.ErrShortSrc {
-				return nil, err
-			}
-			logp.Info("Initialising encoding for '%v' failed due to file being to short")
+		encoding, err = h.encoding(file)
+		if err != nil {
+			logp.Err("Initialising encoding for '%v' failed: %v", file, err)
```

Comment:
@urso Can you please have a close look at this one? Instead of iterating now an error is returned for all errors that occur. Before you didn't return an error on ErrShortSrc. What happens now is that next time a harvester is started again and it hopes for a file that is not too short. Does this make sense?